src/meshes/t8code_mesh.jl

```diff
@@ -0,0 +1,383 @@
+                   mapping, RealT=Float64, initial_refinement_level=0, periodicity=true, unsaved_changes=true)
+
+  NDIMS = length(trees_per_dimension)
+
+  @assert NDIMS == 2 # Only support for NDIMS = 2 yet.
+
+  # Convert periodicity to a Tuple of a Bool for every dimension
+  if all(periodicity)
+    # Also catches case where periodicity = true
+    periodicity = ntuple(_->true, NDIMS)
+  elseif !any(periodicity)
+    # Also catches case where periodicity = false
+    periodicity = ntuple(_->false, NDIMS)
+  else
+    # Default case if periodicity is an iterable
+    periodicity = Tuple(periodicity)
+  end
+
+  conn = p4est_connectivity_new_brick(trees_per_dimension..., periodicity...)
+  do_partition = 0
+  cmesh = t8_cmesh_new_from_p4est(conn,t8_mpi_comm(),do_partition)
+  p4est_connectivity_destroy(conn)
+
+  scheme = t8_scheme_new_default_cxx()
+  forest = t8_forest_new_uniform(cmesh,scheme,initial_refinement_level,0,mpi_comm().val)
+
+  basis = LobattoLegendreBasis(RealT, polydeg)
+  nodes = basis.nodes
+
+  tree_node_coordinates = Array{RealT, NDIMS+2}(undef, NDIMS,
+                                                ntuple(_ -> length(nodes), NDIMS)...,
+                                                prod(trees_per_dimension))
+
+  # Get cell length in reference mesh: Omega_ref = [-1,1]^2.
+  dx = 2 / trees_per_dimension[1]
+  dy = 2 / trees_per_dimension[2]
+
+  num_local_trees = t8_cmesh_get_num_local_trees(cmesh)
+
+  # Non-periodic boundaries.
+  boundary_names = fill(Symbol("---"), 2 * NDIMS, prod(trees_per_dimension))
+
+  for itree = 1:num_local_trees
+    veptr = t8_cmesh_get_tree_vertices(cmesh, itree-1)
+    verts = unsafe_wrap(Array,veptr,(3,1 << NDIMS))
+
+    # Calculate node coordinates of reference mesh.
+    cell_x_offset = (verts[1,1] - 1/2*(trees_per_dimension[1]-1)) * dx
+    cell_y_offset = (verts[2,1] - 1/2*(trees_per_dimension[2]-1)) * dy
+
+    for j in eachindex(nodes), i in eachindex(nodes)
+      tree_node_coordinates[:, i, j, itree] .= mapping(cell_x_offset + dx * nodes[i]/2,
+                                                       cell_y_offset + dy * nodes[j]/2)
+    end
+
+    if !periodicity[1]
+      boundary_names[1, itree] = :x_neg
+      boundary_names[2, itree] = :x_pos
+    end
+
+    if !periodicity[2]
+      boundary_names[3, itree] = :y_neg
+      boundary_names[4, itree] = :y_pos
+    end
+  end
+
+  return T8codeMesh{NDIMS}(cmesh, scheme, forest, tree_node_coordinates, nodes, boundary_names, "", unsaved_changes)
+
+end
+
+"""
+    T8codeMesh{NDIMS}(cmesh::Ptr{t8_cmesh},
+                     mapping=nothing, polydeg=1, RealT=Float64,
+                     initial_refinement_level=0, unsaved_changes=true)
+Main mesh constructor for the `T8codeMesh` that imports an unstructured, conforming
+mesh from an Abaqus mesh file (`.inp`). Each element of the conforming mesh parsed
+from the `meshfile` is created as a [`p4est`](https://github.com/cburstedde/p4est)
+tree datatype.
```

Comment:
This may be a little bit confusing since this should construct a `t8code` mesh, not a `p4est` mesh.